lib/jelix/core/response/jResponseStreamed.class.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * @package     jelix
+ * @subpackage  core_response
+ *
+ * @author      René-Luc Dhont
+ *
+ * @copyright   2021 René-Luc Dhont
+ *
+ * @see        http://www.jelix.org
+ * @licence     GNU Lesser General Public Licence see LICENCE file or http://www.gnu.org/licenses/lgpl.html
+ */
+
+/**
+ * Streamed response.
+ *
+ * Response use to send a streamed data to the client. It
+ * uses a callback for its content.
+ *
+ * The callback should use the standard PHP functions like echo
+ * to stream the response back to the client. The flush() function
+ * can also be used if needed.
+ *
+ * @package  jelix
+ * @subpackage core_response
+ *
+ * @see jResponse
+ * @since 1.8a2
+ */
+final class jResponseStreamed extends jResponse
+{
+    /**
+     * @var string
+     */
+    protected $_type = 'streamed';
+
+    /**
+     * The mimeType of the current streamed response.
+     * It will be sent in the header "Content-Type".
+     *
+     * @var string
+     */
+    public $mimeType = 'application/octet-stream';
+
+    /**
+     * @var callable The callback use to send the content
+     */
+    protected $callback;
+
+    /**
+     * @var bool The content is streamed
+     */
+    protected $streamed;
+
+    /**
+     * @var bool The header has been sent
+     */
+    private $headersSent;
```

Comment:
There is already a `_httpHeadersSent` property that exists on the jResponse class, and modified by the `sendHttpHeaders()` method.  You should remove it.